ext/standaloneusers/Civi/Standalone/Security.php

```diff
@@ -225,12 +225,22 @@ public function authenticate($name, $password, $loadCMSBootstrap = FALSE, $realP
    * Currently only used by CRM_Utils_System_Standalone::loadBootstrap
    */
   public function loginAuthenticatedUserRecord(array $user, bool $withSession) {
-    $authX = new \Civi\Authx\Standalone();
+    global $loggedInUserId, $loggedInUser;
+    $loggedInUserId = $user['id'];
+    $loggedInUser = \Civi\Standalone\Security::singleton()->loadUserByID($user['id']);
+
     if ($withSession) {
-      $authX->loginSession($user['id']);
-    }
-    else {
-      $authX->loginStateless($user['id']);
+      $session = \CRM_Core_Session::singleton();
+      $session->set('ufID', $user['id']);
+
+      // Identify the contact
+      $contactID = civicrm_api3('UFMatch', 'get', [
+        'sequential' => 1,
+        'return' => ['contact_id'],
+        'uf_id' => $user['id'],
+      ])['values'][0]['contact_id'] ?? NULL;
+      // Confusingly, Civi stores it's *Contact* ID as *userId* on the session.
+      $session->set('userId', $contactID);
```

Comment:
Hmm, this expects `userID`: https://github.com/civicrm/civicrm-core/blob/468bbc73b9a2c546842955f7fc0cfec5fafc9c22/CRM/Core/Session.php#L553

Reply:
Good catch @demeritcowboy. FWIW, that's how it was before the (REF), and I guess the bug is sorta distinct. But it's easier to throw the patch on top here. (*The area is under heavy dev, so I don't think we need separate issue for it.*)

Reply:
@artfulrobot Yeah, I agree it's a subculture-vs-subculture thing. So it's not so simple as "always":

* For civicrm-core, the relative proportions are: `camelId` 47% (16 ksloc) and `camelID` 53% (18 ksloc).
* Civi style guidelines don't say much about naming in general (or this in particular).
* There's a general idea to defer to Drupal style (*as long as it doesn't force major breaks/renames*). 
* Drupal doesn't officially speak to this either. However, empirically, Drupal is using `camelId` (93%; 7 ksloc) over `camelID` (7%; <1 ksloc). So they're more consistent.
* But overall volume of camelized identifiers is lower on Drupal (8 ksloc) than Civi (34 ksloc).

Fun links:

* https://softwareengineering.stackexchange.com/questions/186277/should-the-variable-be-named-id-or-id
* https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions (*This is the only style guideline I saw which even tried to do detailed guidance for capitalization edge-cases.*)